gamla/functional.py

```diff
@@ -161,6 +161,24 @@ def acompose_left(*funcs):
     return acompose(*reversed(funcs))
 
 
+async def throttled_amap(f, it, limit):
```

Comment:
so this is essentially `amap` composed with `throttle` - consider separating them into two